src/lib.rs

```diff
@@ -161,8 +163,9 @@ impl DecodeReaderBytesBuilder {
         }
         let encoding = self.encoding
             .map(|enc| enc.new_decoder_with_bom_removal());
-        // No need to do BOM detection if we have an explicit encoding.
-        let has_detected = !self.bom_override && encoding.is_some();
+
+        // No need to do BOM detection if we opt out of it or have an explicit encoding.
+        let has_detected = !self.bom_sniffing || !self.bom_override && encoding.is_some();
```

Comment:
Can you add parens here to indicate precedent?

Reply:
Done!